--- make_linux.mak.orig	2004-10-22 17:48:31.000000000 +0200
+++ make_linux.mak	2004-10-22 17:48:45.286049824 +0200
@@ -43,13 +43,13 @@
 GNOMECFLAGS = `pkg-config --cflags gnome-vfs-module-2.0 libgnome-2.0 libgnomeui-2.0`
 GNOMELIBS = `pkg-config --libs gnome-vfs-module-2.0 libgnome-2.0 libgnomeui-2.0`
 
-KDE_LIBS = -L/usr/lib  -L$(QT_HOME)/lib -shared  -lkdecore -lqt -lkparts
-KDE_CFLAGS = -fno-rtti -c -O -I/usr/include/kde -I$(QT_HOME)/include -I$(JAVA_HOME)/include
+KDE_LIBS = -shared -lkdecore -lqt-mt -lkparts
+KDE_CFLAGS = -fno-rtti -c $(OPT) `pkg-config --cflags qt-mt`
 
 # Uncomment for Native Stats tool
 #NATIVE_STATS = -DNATIVE_STATS
 
-MOZILLACFLAGS = -O \
+MOZILLACFLAGS = $(OPT) \
 	-DXPCOM_GLUE=1 \
 	-DMOZILLA_STRICT_API=1 \
 	-fno-rtti \
@@ -59,18 +59,14 @@
 	-Wno-non-virtual-dtor \
 	-fPIC \
 	-I./ \
-	-I$(GECKO_SDK)	\
-	-include $(GECKO_SDK)/mozilla-config.h \
-	-I$(GECKO_SDK)/nspr/include \
-	-I$(GECKO_SDK)/xpcom/include \
-	-I$(GECKO_SDK)/string/include \
-	-I$(GECKO_SDK)/embed_base/include \
-	-I$(GECKO_SDK)/embedstring/include
+	$(JAVA_INC) \
+	-include /usr/include/mozilla/mozilla-config.h \
+	-I/usr/include/nspr/include \
+	-I/usr/include/mozilla/xpcom \
+	-I/usr/include/mozilla/string \
+	-I/usr/include/mozilla/embed_base
 MOZILLALIBS = -shared -Wl,--version-script=mozilla_exports -Bsymbolic \
-	-L$(GECKO_SDK)/embedstring/bin -lembedstring \
-	-L$(GECKO_SDK)/embed_base/bin -lembed_base_s \
-	-L$(GECKO_SDK)/xpcom/bin -lxpcomglue_s -lxpcom \
-	-L$(GECKO_SDK)/nspr/bin -lnspr4 -lplds4 -lplc4
+	-lembedcomponents -lxpcomglue_s -lxpcom -lnspr4 -lplds4 -lplc4
 
 SWT_OBJECTS = swt.o callback.o
 AWT_OBJECTS = swt_awt.o
@@ -80,11 +76,11 @@
 KDE_OBJS = swt.o kde.o kde_stats.o
 MOZILLA_OBJECTS = swt.o xpcom.o xpcom_custom.o xpcom_structs.o xpcom_stats.o
  
-CFLAGS = -O -Wall \
+CFLAGS = $(OPT) -Wall \
 		-DSWT_VERSION=$(SWT_VERSION) \
 		$(NATIVE_STATS) \
 		-DLINUX -DGTK \
-		-I$(JAVA_HOME)/include \
+		$(JAVA_INC) \
 		-fpic \
 		${SWT_PTR_CFLAGS}
 LIBS = -shared -fpic
@@ -98,16 +94,16 @@
 make_swt: $(SWT_LIB) $(SWTPI_LIB)
 
 $(SWT_LIB): $(SWT_OBJECTS)
-	$(LD) $(LIBS) -o $(SWT_LIB) $(SWT_OBJECTS)
+	$(CC) $(LIBS) -o $(SWT_LIB) $(SWT_OBJECTS)
 
 callback.o: callback.c callback.h
 	$(CC) $(CFLAGS) -c callback.c
 
 $(SWTPI_LIB): $(SWTPI_OBJECTS)
-	$(LD) $(LIBS) $(GTKLIBS) -o $(SWTPI_LIB) $(SWTPI_OBJECTS)
+	$(CC) $(LIBS) $(GTKLIBS) -o $(SWTPI_LIB) $(SWTPI_OBJECTS)
 
 swt.o: swt.c swt.h
-	$(CC) $(CFLAGS) -c swt.c
+	$(CC) $(CFLAGS) $(JAVA_INC) -c swt.c
 os.o: os.c os.h swt.h os_custom.h
 	$(CC) $(CFLAGS) $(GTKCFLAGS) -c os.c
 os_structs.o: os_structs.c os_structs.h os.h swt.h
@@ -123,7 +119,7 @@
 make_awt:$(AWT_LIB)
 
 $(AWT_LIB): $(AWT_OBJECTS)
-	$(LD) $(AWT_LIBS) -o $(AWT_LIB) $(AWT_OBJECTS)
+	$(CC) $(AWT_LIBS) -o $(AWT_LIB) $(AWT_OBJECTS)
 
 #
 # Atk lib
@@ -131,7 +127,7 @@
 make_atk: $(ATK_LIB)
 
 $(ATK_LIB): $(ATK_OBJECTS)
-	$(LD) $(LIBS) $(ATKLIBS) -o $(ATK_LIB) $(ATK_OBJECTS)
+	$(CC) $(LIBS) $(ATKLIBS) -o $(ATK_LIB) $(ATK_OBJECTS)
 
 atk.o: atk.c atk.h
 	$(CC) $(CFLAGS) $(ATKCFLAGS) -c atk.c
@@ -148,7 +144,7 @@
 make_gnome: $(GNOME_LIB)
 
 $(GNOME_LIB): $(GNOME_OBJECTS)
-	$(LD) $(LIBS) $(GNOMELIBS) -o $(GNOME_LIB) $(GNOME_OBJECTS)
+	$(CC) $(LIBS) $(GNOMELIBS) -o $(GNOME_LIB) $(GNOME_OBJECTS)
 
 gnome.o: gnome.c 
 	$(CC) $(CFLAGS) $(GNOMECFLAGS) -c gnome.c
@@ -165,7 +161,7 @@
 make_kde: $(KDE_LIB)
 
 $(KDE_LIB): $(KDE_OBJS)
-	$(LD) -o $@ $(KDE_OBJS) $(KDE_LIBS)
+	$(CXX) -o $@ $(KDE_OBJS) $(KDE_LIBS)
 
 kde.o: kde.cpp
 	$(CXX) $(CFLAGS) $(KDE_CFLAGS) -o kde.o kde.cpp
