Index: library/eclipse.c
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.equinox.executable/library/eclipse.c,v
retrieving revision 1.20
--- library/eclipse.c	8 Feb 2007 16:57:15 -0000	1.20
+++ library/eclipse.c	1 Mar 2007 03:44:19 -0000
@@ -330,8 +330,9 @@
 
 /* Define the required VM arguments (all platforms). */
 static _TCHAR*  cp = NULL;
-static _TCHAR*  cpValue = NULL;					
-static _TCHAR** reqVMarg[] = { &cp, &cpValue, NULL };	/* required VM args */
+static _TCHAR*  cpValue = NULL;
+static _TCHAR* sharedConfigurationDir = _T_ECLIPSE("-Dosgi.sharedConfiguration.area=/usr/lib/eclipse/configuration");					
+static _TCHAR** reqVMarg[] = { &sharedConfigurationDir, &cp, &cpValue, NULL };	/* required VM args */
 static _TCHAR** userVMarg  = NULL;	     				/* user specific args for the Java VM  */
 static _TCHAR** eeVMarg = NULL;							/* vm args specified in ee file */
 static int nEEargs = 0;
@@ -915,20 +916,12 @@
  */
 _TCHAR* getProgramDir( )
 {
-	_TCHAR*  ch;
 	_TCHAR*  programDir;
+	_TCHAR* temp = _T_ECLIPSE("/usr/share/eclipse/");
 
-    programDir = malloc( (_tcslen( program ) + 1) * sizeof(_TCHAR) );
-    _tcscpy( programDir, program );
-    ch = _tcsrchr( programDir, dirSeparator );
-	if (ch != NULL)
-    {
-    	*(ch+1) = _T_ECLIPSE('\0');
-   		return programDir;
-    }
-
-    free( programDir );
-    return NULL;
+	programDir = malloc( (_tcslen( temp + 1 )) * sizeof(_TCHAR) );
+	_tcscpy( programDir, temp );
+	return programDir;
 }
 
 static _TCHAR* findStartupJar(){
