--- plugins/org.eclipse.core.resources.linux/src/Makefile.orig	2004-12-29 14:30:14.722860888 +0000
+++ plugins/org.eclipse.core.resources.linux/src/Makefile	2004-12-29 14:30:50.293453336 +0000
@@ -22,8 +22,8 @@
 JDK_INCLUDE = -I ${JAVA_HOME}/include -I ${JAVA_HOME}/include/${OS_TYPE}
 
 core :
-	gcc -fPIC -g -c $(JDK_INCLUDE) -o $(CORE.O) $(CORE.C)
-	gcc -g -shared -Wl,-soname,$(LIB_NAME) -o $(LIB_NAME_FULL) $(CORE.O) -lc
-
+	gcc -fPIC $(CFLAGS) -c $(INC_PATH) $(CORE.C) -o $(CORE.O)
+	gcc $(LDFLAGS) -shared -Wl,-soname,$(LIB_NAME) -o $(LIB_NAME_FULL) $(CORE.O)
+	
 clean :
 	rm *.o
