--- make_linux.mak.orig	2005-06-27 20:53:36.000000000 +0200
+++ make_linux.mak	2005-06-30 22:05:11.000000000 +0200
@@ -50,7 +50,7 @@
 # Uncomment for Native Stats tool
 #NATIVE_STATS = -DNATIVE_STATS
 
-MOZILLACFLAGS = -O \
+MOZILLACFLAGS = $(OPT) \
 	-DXPCOM_GLUE=1 \
 	-DMOZILLA_STRICT_API=1 \
 	-fno-rtti \
@@ -63,7 +63,13 @@
 	-I$(JAVA_HOME)/include \
 	-I$(JAVA_HOME)/include/linux \
 	${GECKO_INCLUDES} \
-	${SWT_PTR_CFLAGS}
+	${SWT_PTR_CFLAGS} \
+	$(JAVA_INC) \
+	-include /usr/include/mozilla/mozilla-config.h \
+	-I/usr/include/nspr \
+	-I/usr/include/mozilla/xpcom \
+	-I/usr/include/mozilla/string \
+	-I/usr/include/mozilla/embed_base
 	
 MOZILLALIBS = -shared -s -Wl,--version-script=mozilla_exports -Bsymbolic ${GECKO_LIBS}
 	
@@ -75,14 +81,15 @@
 GNOME_OBJECTS = swt.o gnome.o gnome_structs.o gnome_stats.o
 MOZILLA_OBJECTS = swt.o xpcom.o xpcom_custom.o xpcom_structs.o xpcom_stats.o
  
-CFLAGS = -O -Wall \
+CFLAGS = $(OPT) -Wall \
 		-DSWT_VERSION=$(SWT_VERSION) \
 		$(NATIVE_STATS) \
 		-DLINUX -DGTK \
 		-I$(JAVA_HOME)/include \
 		-I$(JAVA_HOME)/include/linux \
 		-fpic \
-		${SWT_PTR_CFLAGS}
+		${SWT_PTR_CFLAGS} \
+		$(JAVA_INC)
 LIBS = -shared -fpic -s
 
 
@@ -94,16 +101,16 @@
 make_swt: $(SWT_LIB) $(SWTPI_LIB)
 
 $(SWT_LIB): $(SWT_OBJECTS)
-	$(LD) $(LIBS) -o $(SWT_LIB) $(SWT_OBJECTS)
-
+	$(CC) $(LIBS) -o $(SWT_LIB) $(SWT_OBJECTS)
+	
 callback.o: callback.c callback.h
 	$(CC) $(CFLAGS) -c callback.c
 
 $(SWTPI_LIB): $(SWTPI_OBJECTS)
-	$(LD) $(LIBS) $(GTKLIBS) -o $(SWTPI_LIB) $(SWTPI_OBJECTS)
-
+	$(CC) $(LIBS) $(GTKLIBS) -o $(SWTPI_LIB) $(SWTPI_OBJECTS)
+	
 swt.o: swt.c swt.h
-	$(CC) $(CFLAGS) -c swt.c
+	$(CC) $(CFLAGS) $(JAVA_INC) -c swt.c
 os.o: os.c os.h swt.h os_custom.h
 	$(CC) $(CFLAGS) $(GTKCFLAGS) -c os.c
 os_structs.o: os_structs.c os_structs.h os.h swt.h
@@ -136,7 +143,7 @@
 make_awt:$(AWT_LIB)
 
 $(AWT_LIB): $(AWT_OBJECTS)
-	$(LD) $(AWT_LIBS) -o $(AWT_LIB) $(AWT_OBJECTS)
+	$(CC) $(AWT_LIBS) -o $(AWT_LIB) $(AWT_OBJECTS)
 
 #
 # Atk lib
@@ -144,7 +151,7 @@
 make_atk: $(ATK_LIB)
 
 $(ATK_LIB): $(ATK_OBJECTS)
-	$(LD) $(LIBS) $(ATKLIBS) -o $(ATK_LIB) $(ATK_OBJECTS)
+	$(CC) $(LIBS) $(ATKLIBS) -o $(ATK_LIB) $(ATK_OBJECTS)
 
 atk.o: atk.c atk.h
 	$(CC) $(CFLAGS) $(ATKCFLAGS) -c atk.c
@@ -161,7 +168,7 @@
 make_gnome: $(GNOME_LIB)
 
 $(GNOME_LIB): $(GNOME_OBJECTS)
-	$(LD) $(LIBS) $(GNOMELIBS) -o $(GNOME_LIB) $(GNOME_OBJECTS)
+	$(CC) $(LIBS) $(GNOMELIBS) -o $(GNOME_LIB) $(GNOME_OBJECTS)
 
 gnome.o: gnome.c 
 	$(CC) $(CFLAGS) $(GNOMECFLAGS) -c gnome.c
