--- plugins/org.eclipse.core.filesystem/natives/unix/linux/Makefile.org	2006-04-21 20:11:42.010647000 +0200
+++ plugins/org.eclipse.core.filesystem/natives/unix/linux/Makefile	2006-04-21 20:14:21.592620250 +0200
@@ -20,11 +20,11 @@
 JAVA_HOME = ~/vm/sun142
 
 JDK_INCLUDE = -I ${JAVA_HOME}/include -I ${JAVA_HOME}/include/${OS_TYPE}
-COMMON_INCLUDE = -I include
+COMMON_INCLUDE = -Iinclude
 
 core :
-	gcc -fPIC -g -c $(JDK_INCLUDE) $(COMMON_INCLUDE) -o $(CORE.O) $(CORE.C)
-	gcc -g -shared -Wl,-soname,$(LIB_NAME) -o $(LIB_NAME_FULL) $(CORE.O) -lc
+	gcc -fPIC $(CFLAGS) -c $(INC_PATH) $(COMMON_INCLUDE) -o $(CORE.O) $(CORE.C)
+	gcc $(LDFLAGS) -shared -Wl,-soname,$(LIB_NAME) -o $(LIB_NAME_FULL) $(CORE.O)
 
 clean :
 	rm *.o
